include/openthread/link_metrics.h

```diff
@@ -141,8 +147,9 @@ typedef void (*otLinkMetricsMgmtResponseCallback)(const otIp6Address *aSource, u
  * @param[in]  aCallback            A pointer to a function that is called when Link Metrics report is received.
  * @param[in]  aCallbackContext     A pointer to application-specific context.
  *
- * @retval OT_ERROR_NONE          Successfully sent a Link Metrics query message.
- * @retval OT_ERROR_NO_BUFS       Insufficient buffers to generate the MLE Data Request message.
+ * @retval OT_ERROR_NONE              Successfully sent a Link Metrics query message.
+ * @retval OT_ERROR_NO_BUFS           Insufficient buffers to generate the MLE Data Request message.
+ * @retval OT_ERROR_UNKNOWN_NEIGHBOR  @p aDestination is not link-local or the neighbor is not found.
```

Comment:
Here `OT_ERROR_INVALID_ARGS` might be more appropriate than  `OT_ERROR_UNKNOWN_NEIGHBOR`?
https://github.com/openthread/openthread/blob/6fda0786edab8e821e1647cc4be08b4a7059cf4a/include/openthread/error.h#L156-L159

Better also check that the neighbor is of thread version >= 3 as Thread 1.1 neighbor won't support this feature for sure.

Reply:
`OT_ERROR_INVALID_ARGS` has already been used to represent other argument errors in the following 3 APIs. I suggest that we add a new error that represents neighbor doesn't support 1.2 features. And in the case `aDestination` is not link-local or neighbor is not found, we use `OT_ERROR_UNKNOWN_NEIGHBOR`. Thoughts?